experimental/tinystr_neo/src/ascii.rs

```diff
@@ -62,18 +64,70 @@ impl<const N: usize> TinyAsciiStr<N> {
         self.bytes.iter().position(|x| *x == 0).unwrap_or(N)
     }
 
+    #[inline]
     pub fn is_empty(&self) -> bool {
         self.bytes[0] == 0
     }
 
+    #[inline]
     pub fn as_bytes(&self) -> &[u8] {
         &self.bytes[0..self.len()]
     }
 
+    #[inline]
     pub fn all_bytes(&self) -> &[u8; N] {
         &self.bytes
     }
 
+    #[inline]
+    pub fn is_ascii_alphabetic(&self) -> bool {
+        if N <= 4 {
+            let aligned = Aligned4::from_bytes(&self.bytes);
+            aligned.is_ascii_alphabetic()
+        } else {
+            self.is_ascii_alphabetic_loop()
+        }
+    }
+
+    fn is_ascii_alphabetic_loop(&self) -> bool {
+        for word in self.bytes.iter() {
+            let mask = (word + 0x7f) & 0x80;
+            let lower = word | 0x20;
+            let alpha = !(lower + 0x1f) | (lower + 0x05);
+            if (alpha & mask) != 0 {
```

Comment:
thought: removing the branch will likely speed up the loop and allow it to be unrolled

(worth trying out at least)

Reply:
I wrote this a different way.